android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java

```diff
@@ -323,7 +323,10 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
           try {
             new URL(urlStr);
           } catch (MalformedURLException e) {
-            KMLog.LogException(TAG, "", e);
+            // Intentionally not logging "No protocol" exception because it may be a query
+            if (e != null && !e.getMessage().startsWith("no protocol:")) {
+              KMLog.LogException(TAG, "", e);
+            }
 
             if (Patterns.WEB_URL.matcher(String.format("%s%s", "http://", urlStr)).matches()) {
```

Comment:
This subsequent test is inadequate: only treat as URL if throwing `http://` on the front makes it look like a URL according to our `WEB_URL` pattern? That may still throw if `webView` thinks the resulting URL is still invalid. It would be better to loop around `new URL(urlStr)` after modifying `urlStr` to add `http://` on the front, i.e. something like this:

```java
bool valid = false;
string originalUrl = urlStr;
while(!valid) {
  try {
    new URL(urlStr);
    valid = true;
  } catch (MalformedURLException e) {
    if (Patterns.WEB_URL.matcher("http://" + urlStr).matches()) {
      urlStr = "http://" + originalUrl;
    } else {
      urlStr = String.format("https://www.google.com/search?q=%s", encodeUriComponent(originalUrl));
      valid = true; // no need to test again; we'll assume it's valid this time
    }
  }
}
```

* Where `encodeUriComponent` is the Java equivalent (which I don't know off the top of my head!) -- this was a bug before.
* Note not using `String.format` (overkill!) for simple string concatenation